File: ChatSupport/src/main/java/com/vantagecircle/chatapp/adapter/UsersMAdapter.java
package com.vantagecircle.chatapp.adapter;

import com.firebase.ui.database.FirebaseRecyclerAdapter;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.Query;
import com.vantagecircle.chatapp.holder.UserMViewHolder;
import com.vantagecircle.chatapp.adapter.interfaceA.ClickUser;
import com.vantagecircle.chatapp.model.UserM;

/**
 * Created by bapidas on 21/07/17.
 */

public class UsersMAdapter extends FirebaseRecyclerAdapter<UserM, UserMViewHolder> {
    private ClickUser clickUser;

    public UsersMAdapter(Class<UserM> modelClass, int modelLayout, Class<UserMViewHolder> viewHolderClass,
                         Query ref, ClickUser clickUser) {
        super(modelClass, modelLayout, viewHolderClass, ref);
        this.clickUser = clickUser;
    }

    public UsersMAdapter(Class<UserM> modelClass, int modelLayout, Class<UserMViewHolder> viewHolderClass,
                         DatabaseReference ref, ClickUser clickUser) {
        super(modelClass, modelLayout, viewHolderClass, ref);
        this.clickUser = clickUser;
    }

    @Override
    protected void populateViewHolder(UserMViewHolder viewHolder, UserM model, int position) {
        viewHolder.setViewHolder(model, clickUser);
    }
}

File: app/src/main/java/com/bapidas/chattingapp/ui/adapter/UsersMAdapter.kt
package com.bapidas.chattingapp.ui.adapter

import com.bapidas.chattingapp.data.model.UserM
import com.bapidas.chattingapp.ui.adapter.callbacks.ClickUser
import com.bapidas.chattingapp.ui.adapter.holder.UserMViewHolder
import com.firebase.ui.database.FirebaseRecyclerAdapter
import com.google.firebase.database.DatabaseReference
import com.google.firebase.database.Query

/**
 * Created by bapidas on 21/07/17.
 */
class UsersMAdapter(modelClass: Class<UserM>, modelLayout: Int,
                    viewHolderClass: Class<UserMViewHolder>,
                    ref: Query?,
                    private var clickUser: ClickUser) :
        FirebaseRecyclerAdapter<UserM, UserMViewHolder>(modelClass, modelLayout, viewHolderClass, ref) {

    override fun populateViewHolder(viewHolder: UserMViewHolder, model: UserM?, position: Int) {
        model?.let { viewHolder.setViewHolder(it, clickUser) }
    }
}